src/deployment/registration.py

```diff
@@ -147,140 +128,135 @@ def register_application(
     subscription_id: str,
 ) -> ApplicationInfo:
     logger.info("retrieving the application registration %s" % registration_name)
-    client = get_graph_client(subscription_id)
-    apps: List[Application] = list(
-        client.applications.list(filter="displayName eq '%s'" % registration_name)
+
+    app = get_application(
+        display_name=registration_name, subscription_id=subscription_id
```

Comment:
Moving the search into `get_application()` makes things much more readable! We might want a `find_` or `query_` function that returns a list (see other comment), but this is a good move overall.